python/ml4ir/base/config/parse_args.py

```diff
@@ -1,10 +1,47 @@
-from argparse import ArgumentParser, Namespace
+from argparse import ArgumentParser, Namespace, Action
 from typing import List
 
 
+class CustomFeatureDictUpdater(Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        """
+        Add a value to a custom argument dictionary.
+        If dictionary does not exist a new one is created.
+        If dictionary already exists, it is updated with new key-value pair.
+
+        Can be used to collect all custom arguments for feature_config
+        by specifying as feature_config.custom_arg 128. This will update the
+        dictionary for feature_config custom args with the new key "custom_arg"
+        and corresponding value "128".
+
+        Parameters
+        ----------
+        parser: ArgumentParser
+            ArgumentParser to use this action with
+        namespace: Namespace
+            The namespace object that will be updated with the parsed value
+        values: str
+            Value to be added to dictionary
+        option_string: str
+            Name of command line argument the action is used with
+
+        Returns
+        -------
+        dict
+            Dictionary with the updated key value pair
+        """
+        old_custom_args = getattr(namespace, self.dest)
```

Comment:
Should we add a check here for the first part of the argument to be only `feature_config` or `model_config`? E.g., return an error when I pass `config.units  52`?

Reply:
I can throw an error here, yes. Let me add that to my current PR.